src/runtime_src/core/pcie/linux/device_linux.cpp

```diff
@@ -67,6 +67,22 @@ struct sysfs_fcn
     return value;
   }
 
+  static ValueType
+  get(const pdev& dev, const char* subdev, const char* entry, const boost::any& v)
+  {
+    std::string err;
+    ValueType value;
+    std::pair <const char*,const char*> p = boost::any_cast<std::pair <const char*,const char*>>(v);
+    if (!strcmp(p.first,"subdev"))
+      subdev = p.second;
+    else if (!strcmp(p.first,"entry"))
+      entry = p.second;
+    dev->sysfs_get(subdev, entry, err, value, static_cast<ValueType>(-1));
+    if (!err.empty())
+      throw std::runtime_error(err);
+    return value;
+  }
```

Comment:
I am not understanding this function  Why is 'v' a std::pair? and why does it contain subdev and entry?  Can you show me where  and how this function ends up getting called through device_query?

The function will not work, because it it assumes 'v' is std::pair, but we will have cases where v is any extra argument needed in the actual implementation of the query request.   The type of arguments needs to be driven from the query structs.

Please explain what you are trying to accomplish and maybe I can assist in a right design.     I don't want to guess, but maybe you are after some dynamic device_query where you indirectly retrieve the subdev and entry.  If this is the case, you need some way to translate the dynamic subdev/entry into a query request and call the standard device_query with the struct.    If the subdev/entry are arbitrary and cannot have its own query struct, then we have to come up with a different mechanism.    I will help, but first help me understand :-)

As FYI, I would like to be reviewer of any PR and involves changes to the infrastructure.